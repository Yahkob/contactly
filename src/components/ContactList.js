import React from 'react';
import Proptypes from 'prop-types';
import Contact from './Contact';
import './ContactList.css';

const ContactList = ({ list }) => (
  <div className="contacts">
    {!list && 'Loading...'}
    {list && list.map(contact => <Contact key={contact.id} contact={contact} />)}
  </div>
);

ContactList.Proptypes = {
  list: Proptypes.array
};

export default ContactList;
