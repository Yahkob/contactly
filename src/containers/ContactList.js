import { connect } from 'react-redux';
import ContactList from '../components/ContactList';

const mapStateToProps = state => ({ list: state.contacts });

const ContactListContainer = connect(mapStateToProps)(ContactList);

export default ContactListContainer;
