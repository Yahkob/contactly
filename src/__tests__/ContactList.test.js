import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import ContactList from '../components/ContactList';
import MockData from '../contacts.json';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <ContactList list={MockData} />
      </MemoryRouter>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
