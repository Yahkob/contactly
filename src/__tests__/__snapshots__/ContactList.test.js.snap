// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="contacts"
>
  <div
    className="contact"
  >
    <a
      href="1"
      onClick={[Function]}
    >
      <div
        className="name"
      >
        Batman
      </div>
    </a>
  </div>
  <div
    className="contact"
  >
    <a
      href="2"
      onClick={[Function]}
    >
      <div
        className="name"
      >
        Jake Harclerode
      </div>
    </a>
  </div>
  <div
    className="contact"
  >
    <a
      href="3"
      onClick={[Function]}
    >
      <div
        className="name"
      >
        Elon Musk
      </div>
    </a>
  </div>
  <div
    className="contact"
  >
    <a
      href="5"
      onClick={[Function]}
    >
      <div
        className="name"
      >
        Head Thornton
      </div>
    </a>
  </div>
  <div
    className="contact"
  >
    <a
      href="6"
      onClick={[Function]}
    >
      <div
        className="name"
      >
        Earnestine Cortez
      </div>
    </a>
  </div>
  <div
    className="contact"
  >
    <a
      href="7"
      onClick={[Function]}
    >
      <div
        className="name"
      >
        Rodriguez Slater
      </div>
    </a>
  </div>
  <div
    className="contact"
  >
    <a
      href="8"
      onClick={[Function]}
    >
      <div
        className="name"
      >
        Brandi Gutierrez
      </div>
    </a>
  </div>
  <div
    className="contact"
  >
    <a
      href="9"
      onClick={[Function]}
    >
      <div
        className="name"
      >
        Day Fields
      </div>
    </a>
  </div>
</div>
`;
